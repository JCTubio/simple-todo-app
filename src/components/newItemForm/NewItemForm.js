import React from 'react'
import { Link } from 'react-router-dom'

export default function NewItemForm({
  title,
  description,
  handleTitleChange,
  handleDescriptionChange,
  handleSubmitForm,
}) {
  return (
    <div className='new-item-form'>
      <label htmlFor='title'>Title:</label>
      <input type='text' onChange={handleTitleChange} value={title} />
      <label htmlFor='description'>Description:</label>
      <textarea
        rows='2'
        name='description'
        onChange={handleDescriptionChange}
        value={description}
      />
      <Link to='/tasks'>
        <button> Back</button>
      </Link>
      <button onClick={handleSubmitForm} value='Submit'>
        Submit
      </button>
    </div>
  )
}
