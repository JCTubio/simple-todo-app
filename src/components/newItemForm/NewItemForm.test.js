import React from "react";
import NewItemForm from "./NewItemForm";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";

describe("The new item form", () => {
  describe("The presentational container", () => {
    it("renders as expected", () => {
      function defaultHandler() {}
      const tree = renderer
        .create(
          <MemoryRouter>
            <NewItemForm
              title="Title"
              description="Description"
              handleTitleChange={defaultHandler}
              handleDescriptionChange={defaultHandler}
              handleSubmitForm={defaultHandler}
              isShowingSubmissionError={false}
            />
          </MemoryRouter>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
