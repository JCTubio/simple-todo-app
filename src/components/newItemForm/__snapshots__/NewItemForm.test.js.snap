// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The new item form The presentational container renders as expected 1`] = `
<div
  className="item-form"
>
  <h1>
    New task
  </h1>
  <div
    className="task-title"
  >
    <label
      htmlFor="title"
    >
      Title:
      <span
        style={
          Object {
            "color": "lightcoral",
          }
        }
      >
        *
      </span>
    </label>
    <input
      className=""
      onChange={[Function]}
      placeholder="Your awesome title goes here!"
      type="text"
      value="Title"
    />
  </div>
  <div
    className="task-description"
  >
    <label
      htmlFor="description"
    >
      Description:
    </label>
    <textarea
      name="description"
      onChange={[Function]}
      placeholder="And here you should provide a great description for your task!"
      rows="2"
      value="Description"
    />
  </div>
  <a
    href="/tasks"
    onClick={[Function]}
  >
    <button
      className="back-button"
    >
       Back
    </button>
  </a>
  <button
    className="submit-button"
    onClick={[Function]}
    value="Submit"
  >
    Submit
  </button>
</div>
`;
